src/tr2/game/objects/creatures/yeti.c

```diff
@@ -32,3 +93,233 @@ void Yeti_Setup(void)
+        }
+        Creature_Mood(item, &info, MOOD_ATTACK);
+        angle = Creature_Turn(item, creature->maximum_turn);
+
+        switch (item->current_anim_state) {
+        case YETI_STATE_STOP:
+            creature->flags = 0;
+            creature->maximum_turn = 0;
+            if (creature->mood == MOOD_ESCAPE) {
+                item->goal_anim_state = YETI_STATE_RUN;
+            } else if (item->required_anim_state != YETI_STATE_EMPTY) {
+                item->goal_anim_state = item->required_anim_state;
+            } else if (creature->mood == MOOD_BORED) {
+                const int32_t random = Random_GetControl();
+                if (random < YETI_WAIT_1_CHANCE || !lara_alive) {
+                    item->goal_anim_state = YETI_STATE_WAIT_1;
+                } else if (random < YETI_WAIT_2_CHANCE) {
+                    item->goal_anim_state = YETI_STATE_WAIT_2;
+                } else if (random < YETI_WALK_CHANCE) {
+                    item->goal_anim_state = YETI_STATE_WALK;
+                }
+            } else if (
+                info.ahead != 0 && info.distance < YETI_ATTACK_1_RANGE
+                && Random_GetControl() < YETI_ATTACK_1_CHANCE) {
+                item->goal_anim_state = YETI_STATE_ATTACK_1;
+                break;
+            } else if (info.ahead != 0 && info.distance < YETI_ATTACK_2_RANGE) {
+                item->goal_anim_state = YETI_STATE_ATTACK_2;
+            } else if (creature->mood == MOOD_STALK) {
+                item->goal_anim_state = YETI_STATE_WALK;
+            } else {
+                item->goal_anim_state = YETI_STATE_RUN;
+            }
+            break;
+
+        case YETI_STATE_WAIT_1:
+            if (creature->mood == MOOD_ESCAPE || item->hit_status) {
+                item->goal_anim_state = YETI_STATE_STOP;
+            } else if (creature->mood == MOOD_BORED) {
+                if (lara_alive) {
+                    const int32_t random = Random_GetControl();
+                    if (random < YETI_WAIT_1_CHANCE) {
+                        item->goal_anim_state = YETI_STATE_STOP;
+                    } else if (random < YETI_WAIT_2_CHANCE) {
+                        item->goal_anim_state = YETI_STATE_WAIT_2;
+                    } else if (random < YETI_WALK_CHANCE) {
+                        item->goal_anim_state = YETI_STATE_STOP;
+                        item->required_anim_state = YETI_STATE_WALK;
+                    }
+                }
+            } else if (Random_GetControl() < YETI_STOP_ROAR_CHANCE) {
+                item->goal_anim_state = YETI_STATE_STOP;
+            }
+            break;
+
+        case YETI_STATE_WAIT_2:
+            if (creature->mood == MOOD_ESCAPE || item->hit_status) {
+                item->goal_anim_state = YETI_STATE_STOP;
+            } else if (creature->mood == MOOD_BORED) {
+                const int32_t random = Random_GetControl();
+                if (random < YETI_WAIT_1_CHANCE && lara_alive) {
```

Comment:
Just a sanity check as this is `||  !lara_alive` in `YETI_STATE_STOP` above - I can't see any behaviour differences but just thought it worth to double check.

Reply:
Same as above.